nova_vm/src/engine/bytecode/vm.rs

```diff
@@ -100,6 +100,26 @@ impl Vm {
         while let Some(instr) = executable.get_instruction(&mut vm.ip) {
             eprintln!("Executing instruction {:?}", instr.kind);
             match instr.kind {
+                Instruction::BitwiseNot => {
+                    // 2. Let oldValue be ? ToNumeric(? GetValue(expr)).
+                    let old_value = to_numeric(agent, vm.result.take().unwrap())?;
+
+                    // 3. If oldValue is a Number, then
+                    if let Ok(old_value) = Number::try_from(old_value) {
+                        // a. Return Number::bitwiseNOT(oldValue).
+                        vm.result = Some(Number::bitwise_not(agent, old_value)?.into());
+                    }
+                    // 4. Else,
+                    else {
```

Comment:
**nitpick:** This sort of comment-in-the-middle-of-if-else can be really confusing sometimes as it makes the `if` branch seem else-less. I've started just moving the `// 4. Else` comment inside the `else` branch, trading some spec language beauty for (hopefully) more readable code :)